base/task/thread_pool/worker_thread_stack.cc

```diff
@@ -18,6 +19,11 @@ WorkerThreadStack::~WorkerThreadStack() = default;
 
 void WorkerThreadStack::Push(WorkerThread* worker) {
   DCHECK(!Contains(worker)) << "WorkerThread already on stack";
+
+  if (recordreplay::IsRecordingOrReplaying() && !recordreplay::AreEventsDisallowed()) {
```

Comment:
The IsRecordingOrReplaying() call isn't necessary here.  Also, we generally want to specify a feature name when calling IsRecordingOrReplaying(), so that we can easily disable the behavior changes we make based on these calls when diagnosing problems that happen while recording.